contrib/bash/goshy

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -x
+show_help() {
+  cat <<EOF
+Usage: goshy [Options] FILE
+Options:
+  -b --burn              Burn after reading.
+  -h --help              Print this help message.
+  -u --only-url          Print only URL as response.
+  -p --period TIMEFRAME  Set a custom expiry period.
+EOF
+}
+
+while getopts p:period: flag
+do
+  case "$\{flag\}" in
+    p) PERIOD_IS_SET=1;;
+    period) PERIOD_IS_SET=1;;
+  esac
+done > /dev/null 2>&1
+
+if [ -z $PERIOD_IS_SET ]; then
+  PERIOD='3d'
+fi
+
+while :; do
+  case "$1" in
+    -p|--period)
+      PERIOD=$2
+      shift
+      ;;
+
+    -b|--burn)
+      BURN='-F burn=1'
+      ;;
+
+    -u|--only-url)
+      ONLYURL='?onlyURL'
+      ;;
+
+    -h|--help)
+      show_help
+      exit 0
+      ;;
+
+    *)
+      FILE=$1
+      break
+  esac
```

Comment:
Maybe it's a good idea to pre define all those variables before the loop below `PERIOD`. By doing so, we can `set -eu` in the beginning.

Reply:
In a nutshell, this would result in a termination if some variable wasn't defined. Thus, it ensures the presence of all used variables.

https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

Reply:
Got it, will add it soon.